src/securityPolicy.c

```diff
@@ -256,17 +281,20 @@ security_policy_t policyForSignTxInit(
 		break;
 
 	case SIGN_TX_SIGNINGMODE_PLUTUS_TX:
-		WARN_IF(numCollaterals == 0);
-		WARN_UNLESS(includeScriptDataHash);
-		WARN();
+		// no tx body element is forbidden in general
 		break;
 
 	default:
 		ASSERT(false);
 	}
 
-	WARN_IF(networkId != MAINNET_NETWORK_ID && networkId != TESTNET_NETWORK_ID);
-	WARN_IF(protocolMagic != MAINNET_PROTOCOL_MAGIC);
+	// there are separate screens for various warnings
+	// the return value of the policy only says that at least one should be applied
+	// and the need for individual warnings is reassessed in the UI machine
+	WARN_UNLESS(isNetworkUsual(networkId, protocolMagic));
+	WARN_IF(needsRunningScriptWarning(numCollaterals));
+	WARN_IF(needsMissingCollateralWarning(txSigningMode, numCollaterals));
+	WARN_IF(needsMissingScriptDataHashWarning(txSigningMode, includeScriptDataHash));
```

Comment:
Not sure I like these two outside of the signing mode switch, since they are obviously related to that one signing mode.

Reply:
I thought about that quite a lot, but I don't think it is obvious from their name or their use in `signTx.c` that it is only relevant for a single signing mode. It is true for now, but we might later find out that we need more Plutus tx modes or whatever, and keeping them here will withstand such a change better.

Also, the UI state machine displaying warnings does not distinguish between signing modes, and I want to keep the logic deciding which warnings to show as much the same as I can make it.